### ISSUE TITLE
Recurse into nested lists

### ISSUE THREAD

**Issue by SimonSapin** [andreyvit/json-diff#4]
It would be nice do a fine-grained comparison into nested lists, instead of marking the whole list as different.

Test case: diff `[1, [2,3]]` and `[1, [2,42]]`

Current result:

```
 [
   1
-  [
-    2
-    3
-  ]
+  [
+    2
+    42
+  ]
 ]
```

Expected result:

```
 [
   1
   [
     2
-    3
+    42
   ]
 ]
```

**Comment by TimAshton** [comment]
@andreyvit  Would you still accept a PR for this?

**Comment by andreyvit** [comment]
@TimAshton Yes. In fact, I've just added a contribution policy to the README, promising to accept any reasonable pull requests and give out commit access.

**Comment by ewoudenberg** [comment]
This has been fixed in 0.6.0:
```
$ ./node_modules/.bin/json-diff --full <(echo '[1,[2,3]]')  <(echo '[1,[2,42]]')
 [
   1
   [
     2
-    3
+    42
   ]
 ]
```
